src/main/kotlin/org/rust/ide/inspections/import/AutoImportFix.kt

```diff
@@ -325,7 +325,11 @@ class AutoImportFix(element: RsElement) : LocalQuickFixOnPsiElement(element), Hi
                 .createPathInTmpMod(importingPathName, context.mod, context.ns, info.usePath, externCrateName) ?: return false
             val element = path.reference.deepResolve() as? RsQualifiedNamedElement ?: return false
             if (!context.namespaceFilter(element)) return false
-            return !(element.parent is RsMembers && element.ancestorStrict<RsTraitItem>() != null)
+            // Looks like it's useless to access trait associated types or constants directly
+            // (i.e. `Trait::CONSTANT`), but associated functions can be used in UFCS
+            val isTraitAssocTypeOrConstant =
```

Comment:
Hm, it's interesting
There is at least one example when I can use assoc constant with `Trait::CONSTANT` syntax
```rust
struct Foo;

impl Bar for Foo {
    const C: Self = Foo;
}

pub trait Bar {
    const C: Self;
}

fn main() {
    let f: Foo = Bar::C;
}
```

Reply:
Wow, nice case. Fixed